### PR TITLE
Add danger

### PR DESCRIPTION
**Pull request by hawkrives**
Danger (http://danger.systems/) is a bot for linting PRs.

> Danger runs during your CI process, and gives teams the chance to automate common code review chores.
>
> This provides another logical step in your build, through this Danger can help lint your rote tasks in daily code review.
>
> You can use Danger to codify your teams norms. Leaving humans to think about harder problems.
> 
> She does this by leaving messages inside your PRs based on rules that you create.
> 
> Over time, as rules are adhered to, the message is amended to reflect the current state of the code review.

---

In our case, I have her looking for a few things:

- new/modified JS files that don't have `@flow` at the top
- new/modified tests that have `it.only` – since that stops any other tests from running
- when a PR goes over 400 lines of code

and she can be extended to look for other stuff as time goes on.

### CODE REVIEW

File: dangerfile.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+})
+
+if (unFlowedFiles.length > 0) {
+  warn(`These new JS files do not have Flow enabled: ${unFlowedFiles.join(", ")}`)
+}
+
+const packageChanged = includes(danger.git.modified_files, 'package.json');
+const lockfileChanged = includes(danger.git.modified_files, 'yarn.lock');
+if (packageChanged && !lockfileChanged) {
+	  const message = 'Changes were made to package.json, but not to yarn.lock';
```
Comment (drewvolz) [inline dangerfile.js]: I'll fix the tabs/spacing issues over lines 19-21

**Review by drewvolz — APPROVED** [review]
Looks good to me.

**Comment by hawkrives** [comment]
One cool thing about Danger is that it only ever posts one comment, and updates / removes that comment as subsequent builds pass.

**Review by elijahverdoorn — APPROVED** [review]
More CI! More Tests!

**Pull request merged**